lib/GHCup/Utils/Pager.hs

```diff
@@ -0,0 +1,54 @@
+{-# LANGUAGE CPP               #-}
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE RankNTypes  #-}
+
+module GHCup.Utils.Pager where
+
+import GHCup.Prelude
+import System.Environment
+import GHCup.Utils.Dirs (findExecutable)
+import Control.Applicative ((<|>))
+import Conduit
+import Data.Conduit.Binary
+import System.Process
+import System.Exit
+import System.IO
+import Data.ByteString (ByteString)
+import Data.Text (Text)
+
+import qualified Data.ByteString.Lazy as BSL
+import qualified Data.Text.Encoding   as E
+
+getPager :: IO (Maybe FilePath)
+getPager
+  | isWindows = pure Nothing
+  | otherwise = do
+      lookupEnv "GHCUP_PAGER" >>= \case
+        Just r  -> pure $ Just r
+        Nothing -> lookupEnv "PAGER" >>= \case
+          Just r' -> pure $ Just r'
+          Nothing -> do
+                findExecutable "most"
+            <|> findExecutable "more"
+            <|> findExecutable "less"
+
+sendToPager :: FilePath -> Text -> IO ()
+sendToPager pager text = sendToPagerConduit pager (sourceLbs (BSL.fromStrict . E.encodeUtf8 $ text))
+
+sendToPagerConduit :: FilePath -> ConduitT () ByteString (ResourceT IO) () -> IO ()
+sendToPagerConduit pager producer = do
+     ((Just stdinH),_,(Just stderrH),ph) <-
+       createProcess
```

Comment:
I am not confident about this, for a start using `withCreateProcess` might be better. And should the std_out be `Inherit`, and delegate_ctlc = True?

I had looked at the implementation done by git and it is doing signal handling changes, like ignoring signals which ought to be handled by pager.